modules/sops/default.nix

```diff
@@ -168,9 +213,18 @@ in {
 
     system.activationScripts.setup-secrets = let
       sops-install-secrets = (pkgs.callPackage ../.. {}).sops-install-secrets;
-    in stringAfter [ "specialfs" "users" "groups" ] ''
+    in stringAfter ([ "specialfs" "users" "groups" ] ++ optional cfg.age.generateKey "generate-age-key") ''
       echo setting up secrets...
-      ${optionalString (cfg.gnupgHome != null) "SOPS_GPG_EXEC=${pkgs.gnupg}/bin/gpg"} ${sops-install-secrets}/bin/sops-install-secrets ${checkedManifest}
+      ${optionalString (cfg.gnupg.home != null) "SOPS_GPG_EXEC=${pkgs.gnupg}/bin/gpg"} ${sops-install-secrets}/bin/sops-install-secrets ${checkedManifest}
     '';
+
+    system.activationScripts.generate-age-key = (mkIf cfg.age.generateKey) (stringAfter [] ''
```

Comment:
`TODO`: I don't think this option is documented yet in Readme.

Reply:
Looks like I rebased that in at some point